### DISCUSSION TITLE
Welcome to CirculatorySystemModels.jl Discussions!

### DISCUSSION THREAD

**Discussion by TS-CUBED** [TS-CUBED/CirculatorySystemModels.jl#20] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kylebeggs** [comment]
Greetings, I'm a PhD candidate in computational mechanics and software engineer for an optics company both using Julia. My research mainly involves the use of meshless methods for solution of PDEs, but I am using hemodynamics as a test bed so this package is relevant as I am using these circuits / 0D models / lumped parameter models as the boundary conditions for the 2D/3D hemodynamics domains. I have already done this ([A meshless multiscale method for simulating hemodynamics](https://www.sciencedirect.com/science/article/abs/pii/S095579972300036X)) but in a much more crude manner, although still using DifferentialEquations.jl. The ModelingToolkit.jl package is great and I'm very happy to see a package using it for reduced order hemodynamics.

There are a few ideas/suggestions I have:
1. Break up the main module file into smaller / more manageable source files. Perhaps we can group all types of resistors into one file, heart models into one file, etc. This includes the `test` folder as well. Typically you would mirror the `src` folder when it makes sense to do so.
2. Provide more convenient functions to plot all or a subset of the variables. I think ModelingToolkit.jl may already have some of this functionality?
3. Allow for users to input custom waveforms either as an input to the system or the output of the system and then do the inverse problem to find parameter values.

Looking forward to some development with you all!

**Comment by TS-CUBED** [comment]
Thanks Kyle. Need to take a closer look at your work - it looks very interesting indeed.

Thanks for your suggestions.

ad 1: yes, the module needs a bit of a clean up and sorting. That's definitely on my todo list. Tests are a bit trickier, since individual components are tricky to test, so the tests are basically our current benchmark problems - which combine many different elements.

ad 2: the ModlingToolkit plot recipes for acausal models are pretty complete in that sense. All variables (both states and observables) are available using the `idxs` keyword.

ad 3: custom waveforms as input are available for some elements, e.g. the `DrivenFlow` or `DrivenPressure` sources. I could imagine providing a ventricle element that allows an external function as the driver for elastance as well. Inverse problems we left out of this, since we think they are best solved using Optimization.jl or other packages in the SciML framework. This is meant more as a standard library for 0D lumped models of the CVS, and it may, indeed, one day be moved to the ModelingToolkitStandardLibrary.

**Reply by kylebeggs** [reply]
Thanks for the response! Given the goal is to be a "standard" library, I think we should make a task list (or just issues) for any element types left to implement? What are the other major milestones needed to reach maturity to get into the ModelingToolkitStandardLibrary?
